app/labor/rate_limit_checker.rb

```diff
@@ -14,6 +14,8 @@ def limit_by_situation(situation)
                user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
              when "image_upload"
                Rails.cache.read("#{user.id}_image_upload").to_i > 9
+             when "notification_subscriptions"
+               Rails.cache.read("#{user.id}_notification_subscriptions").to_i > 9
```

Comment:
This strikes me as a pretty odd implementation. Any reason not to query for the number of notification subscriptions in the past x minutes here instead of this cache write/read situation? It seems like we have all the info to not need to rely on this.

Reply:
That's a good point. I think in my haste I copied the implementation for `image_upload`. I'll update it now.